lib/hsa/mcwamp_hsa.cpp

```diff
@@ -494,15 +499,18 @@ class HSABarrier : public Kalmar::KalmarAsyncOp {
     bool isDispatched;
     hsa_wait_state_t waitMode;
 
+
     std::shared_future<void>* future;
 
     // prior dependencies
     // maximum up to 5 prior dependencies could be associated with one
     // HSABarrier instance
     int depCount;
 
-
 public:
+    uint16_t  header;
```

Comment:
Also it seems this field `header` is added mostly for debugging purpose, correct?